bin/check-alertclear-patterns.rb

```diff
@@ -0,0 +1,261 @@
+         default: "#{BASE_DIR}/default"
+
+  option :log_file,
+         description: 'Path to log file',
+         short: '-f FILE',
+         long: '--log-file FILE'
+
+  option :pattern_alert,
+         description: 'Pattern to create alert',
+         short: '-A PAT',
+         long: '--pattern-alert PAT'
+
+  option :pattern_clear,
+         description: 'Pattern to clear alert',
+         short: '-C PAT',
+         long: '--pattern-clear PAT'
+
+  option :pattern_identifier,
+         description: 'Pattern regex to extract alert name from alert log line',
+         short: '-I PAT',
+         long: '--pattern-identifier PAT'
+
+  option :encoding,
+         description: 'Explicit encoding page to read log file with',
+         short: '-e ENCODING-PAGE',
+         long: '--encoding ENCODING-PAGE'
+
+  option :only_warn,
+         description: 'Warn instead of critical on match',
+         short: '-o',
+         long: '--warn-only',
+         boolean: true
+
+  option :handler,
+         short: '-h HANDLER',
+         long: '--handler HANDLER',
+         default: 'default'
+
+  option :extra_event_args,
+         description: 'Add extra arguments to the event data created. Format is JSON',
+         short: '-x ARGS',
+         long: '--extra-event-args ARGS'
+
+  option :case_insensitive,
+         description: 'Run a case insensitive match',
+         short: '-i',
+         long: '--icase',
+         boolean: true,
+         default: false
+
+  option :file_pattern,
+         description: 'Check a pattern of files, instead of one file',
+         short: '-F FILE',
+         long: '--filepattern FILE'
+
+  option :static_checkId,
+         description: 'Static content, that is needed on the check_Id',
+         short: '-P STATIC_CHECK_ID',
+         long: '--staticcheckid STATIC_CHECK_ID'
+
+  option :verbose,
+         short: '-v',
+         long: '--verbose',
+         boolean: true,
+         default: false
+
+  option :dry_run,
+         short: '-d',
+         long: '--dry-run',
+         boolean: true,
+         default: false
+
+  def run
+    unknown 'No log file specified' unless config[:log_file] || config[:file_pattern]
+    file_list = []
+    file_list << config[:log_file] if config[:log_file]
+    if config[:file_pattern]
+      dir_str = config[:file_pattern].slice(0, config[:file_pattern].to_s.rindex('/'))
+      file_pat = config[:file_pattern].slice((config[:file_pattern].to_s.rindex('/') + 1), config[:file_pattern].length)
+      Dir.foreach(dir_str) do |file|
+        if config[:case_insensitive]
+          file_list << "#{dir_str}/#{file}" if file.to_s.downcase.match(file_pat.downcase)
+        elsif file.to_s.match(file_pat)
+          file_list << "#{dir_str}/#{file}"
+        end
+      end
+    end
+    file_list.each do |log_file|
+      begin
+        open_log log_file
+      rescue => e
+        unknown "Could not open log file: #{e}"
+      end
+      search_log
+    end
+  end
+
+  def open_log(log_file)
```

Comment:
Could we move this to a class module so it could be used across multiple checks?